app/models/user.rb

```diff
@@ -36,9 +39,10 @@ class User < ApplicationRecord
   validates :dob, timeliness: { type: :date, on_or_before: Date.yesterday }, if: :dob_changed?
```

Comment:
The allow_blank: true option in the email validation contradicts the presence: true validation. If email must be present, allow_blank should be removed to avoid confusion and potential issues.